### ISSUE TITLE
Fix S2190 FN: Consider property setter as recursion

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#3624]
### Description

S2190 triggers infinite recursion warning on the getter. Same warning should be triggered for the setter as well, because the setter is also recursive call.

```C#
public string Name
{
    get // Noncompliant
    {
        return Name;
    }
    set // False Negative
    {
        Name = value;
    }
}
```

Original source: [community thread](https://community.sonarsource.com/t/getters-and-setters-without-private-variable/30719)
